### PR TITLE
[Term Entry] Kotlin/Strings/partition

### LINKED ISSUE #3730: [Term Entry] Kotlin Strings .partition()
### Reason/inspiration (optional)

We would like a new term entry on the `partition()` method for Kotlin: [Strings](https://www.codecademy.com/resources/docs/kotlin/strings). The entry should go in a new file under `docs/content/kotlin/concepts/strings/terms/partition/partition.md`.

The entry should describe what the method does. Be sure to include the following sections:

- A "Syntax" section with a ` ```pseudo` code snippet.
- An "Example" section with a ` ```kotlin `code snippet.

Please refer to the [entry templates](https://github.com/Codecademy/docs/blob/main/documentation/), [content standards](https://github.com/Codecademy/docs/blob/main/documentation/content-standards.md), and [markdown style guide](https://curriculum-documentation.codecademy.com/content-guidelines/markdown-style-guide/) when working on the PR for this issue.

### Entry Type (select all that apply)

New Term for Existing Concept Entry

### Code of Conduct

- [X] By submitting this issue, I agree to follow Codecademy Doc's [Code of Conduct](https://github.com/Codecademy/docs/blob/main/.github/CODE_OF_CONDUCT.md).

### For Maintainers

- [X] Labels added
- [x] Issue is assigned

### PR DESCRIPTION
**Pull request by noahpgordon**
### Description

Added **partition.md** within Kotlin Strings. Closes #3730 

### Type of Change

- Adding a new entry

### Checklist

- [X] All writings are my own.
- [X] My entry follows the Codecademy Docs style guide.
- [X] My changes generate no new warnings.
- [X] I have performed a self-review of my own writing and code.
- [X] I have checked my entry and corrected any misspellings.
- [X] I have made corresponding changes to the documentation if needed.
- [X] I have confirmed my changes are not being pushed from my forked `main` branch.
- [X] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [X] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

File: content/kotlin/concepts/strings/terms/partition/partition.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+A predicate takes an argument and returns a boolean `true` or `false`. If `true` is returned, the argument is said to *satisfy* the predicate. Otherwise, the argument does not satisfy the predicate.
+
+## Syntax
+
+```psuedo
+xString.partition(xPredicate)
+```
+
+Where `xString` supplies the string to be partitioned by the predicate `xPredicate`.
```
Comment (dakshdeepHERE) [inline content/kotlin/concepts/strings/terms/partition/partition.md:26]:
```suggestion
- **xString:** This supplies the string to be partitioned by the predicate.
```

File: content/kotlin/concepts/strings/terms/partition/partition.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
```
Comment (dakshdeepHERE) [inline content/kotlin/concepts/strings/terms/partition/partition.md:27]: explain here what is `xPradicate like you did for the `xString` above

File: content/kotlin/concepts/strings/terms/partition/partition.md [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+The following example uses the character method `.isUpperCase()` to partition a string into uppercase letter characters and other characters:
+
+```kotlin
+fun main () {
+  val sampleString = "AbCd12/"
+  println(sampleString.partition(){it.isUpperCase()})
+}
+```
+
+This outputs:
```
Comment (dakshdeepHERE) [inline content/kotlin/concepts/strings/terms/partition/partition.md:39]:
```suggestion
The above example results in the following output:
```

**Review by dakshdeepHERE — CHANGES_REQUESTED** [review]
I have few comments for you @noahpgordon. 🚀

File: content/kotlin/concepts/strings/terms/partition/partition.md [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+```psuedo
+xString.partition(xPredicate)
+```
+
+Where `xString` supplies the string to be partitioned by the predicate `xPredicate`.
+
+## Example
+
+The following example uses the character method `.isUpperCase()` to partition a string into uppercase letter characters and other characters:
```
Comment (KTom101) [inline content/kotlin/concepts/strings/terms/partition/partition.md:30]:
```suggestion
The following example uses `.partition()` and the character method `.isUpperCase()` to partition a string into uppercase letter characters and other characters:
```

**Comment by noahpgordon** [comment]
@dakshdeepHERE @KTom101 I've made relevant edits.

**Review by dakshdeepHERE — APPROVED** [review]
Looks good for second review

**Pull request merged**